### PR TITLE
Remove `-alpha` OpenTelemetry dependencies

### LINKED ISSUE #1232: NoClassDefFoundError: io/opentelemetry/semconv/HttpAttributes
Hi

Yesterday I updated the msgraph-sdk-java from v6.5.1 to v6.6.0, which updates kiota from 1.1.2 to 1.1.4.

Since then I get the following exception:
```
2024-04-17T19:00:06.002288463+02:00 | ERROR | Error occurred while executing task for trigger IntervalTrigger [id=1_my.corp.directory.DirectoryReader#updateUsers, interval=900000] | : java.util.concurrent.CompletionException: java.lang.NoClassDefFoundError: io/opentelemetry/semconv/HttpAttributes
        at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:332)
        at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:347)
        at java.base/java.util.concurrent.CompletableFuture.uniWhenComplete(CompletableFuture.java:874)
        at java.base/java.util.concurrent.CompletableFuture.uniWhenCompleteStage(CompletableFuture.java:887)
        at java.base/java.util.concurrent.CompletableFuture.whenComplete(CompletableFuture.java:2357)
        at java.base/java.util.concurrent.CompletableFuture$MinimalStage.whenComplete(CompletableFuture.java:2948)
        at io.quarkus.scheduler.common.runtime.DefaultInvoker.invoke(DefaultInvoker.java:24)
        at io.quarkus.scheduler.common.runtime.StatusEmitterInvoker.invoke(StatusEmitterInvoker.java:35)
        at io.quarkus.scheduler.runtime.SimpleScheduler$ScheduledTask.doInvoke(SimpleScheduler.java:443)
        at io.quarkus.scheduler.runtime.SimpleScheduler$ScheduledTask$2.call(SimpleScheduler.java:425)
        at io.quarkus.scheduler.runtime.SimpleScheduler$ScheduledTask$2.call(SimpleScheduler.java:422)
        at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$0(ContextImpl.java:177)
        at io.vertx.core.impl.ContextInternal.dispatch(ContextInternal.java:276)
        at io.vertx.core.impl.ContextImpl.lambda$internalExecuteBlocking$2(ContextImpl.java:209)
        at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:587)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
        at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
        at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.NoClassDefFoundError: io/opentelemetry/semconv/HttpAttributes
        at com.microsoft.kiota.http.OkHttpRequestAdapter.getRequestFromRequestInformation(OkHttpRequestAdapter.java:878)
        at com.microsoft.kiota.http.OkHttpRequestAdapter.getHttpResponseMessage(OkHttpRequestAdapter.java:719)
        at com.microsoft.kiota.http.OkHttpRequestAdapter.send(OkHttpRequestAdapter.java:280)
        at com.microsoft.graph.groups.item.members.MembersRequestBuilder.get(MembersRequestBuilder.java:146)
        at my.corp.directory.DirectoryReader.updateUsers(DirectoryReader.java:163)
        at my.corp.directory.DirectoryReader_ClientProxy.updateUsers(Unknown Source)
        at my.corp.directory.DirectoryReader_ScheduledInvoker_updateUsers_fa922a3d5993a039372a4e6b393ec8711b9c46e0.invokeBean(Unknown Source)
        ... 15 more
Caused by: java.lang.ClassNotFoundException: io.opentelemetry.semconv.HttpAttributes
        ... 22 more
```
It runs inside Quarkus in a Scheduler. Do you have any hints on how I could fix this?

Maybe it's related to this change? https://github.com/microsoft/kiota-java/commit/a191db91b89bd7b08699b88accd9408e1487032d

### PR DESCRIPTION
**Pull request by andreaTP**
Fixes #1232

### CODE REVIEW

**Review by andrueastman — DISMISSED** [review]

**Comment by baywet** [comment]
@andreaTP thanks for putting this together. My only concern here is that keys might change over time and we've now traded a compile time issue for a runtime issue harder to investigate.
I think it's fair to assume that the stable attributes won't change, but the experimental ones might.
Could we move all the strings to constants, amend which ones are experimental and which ones are stable, add a link to this documentation https://opentelemetry.io/docs/specs/semconv/attributes-registry/http/ and potentially add a note to replace the constants by the dependencies in the future when they become stable?

**Comment by andreaTP** [comment]
> My only concern here is that keys might change over time and we've now traded a compile time issue for a runtime issue harder to investigate.

This is not accurate, the reported error is a runtime issue that crashes the user application.
After this change, a possible mismatch will result in erroneous keys used, probably hard to spot I agree, but safe from the code execution standpoint.

> Could we move all the strings to constants ...

I agree this will be good, I'll try to review all the attributes that are being used.

File: components/http/okHttp/src/main/java/com/microsoft/kiota/http/OkHttpRequestAdapter.java [hunk 1]
```diff
@@ -874,15 +869,14 @@ private void setBaseUrlForRequestInformation(@Nonnull final RequestInformation r
...
+            spanForAttributes.setAttribute(HTTP_REQUEST_METHOD, requestInfo.httpMethod.toString());
             final URL requestURL = requestInfo.getUri().toURL();
             if (obsOptions.getIncludeEUIIAttributes()) {
-                spanForAttributes.setAttribute(UrlAttributes.URL_FULL, requestURL.toString());
+                spanForAttributes.setAttribute(URL_FULL, requestURL.toString());
             }
-            spanForAttributes.setAttribute("http.port", requestURL.getPort());
-            spanForAttributes.setAttribute(ServerAttributes.SERVER_ADDRESS, requestURL.getHost());
-            spanForAttributes.setAttribute(UrlAttributes.URL_SCHEME, requestURL.getProtocol());
+            spanForAttributes.setAttribute(SERVER_PORT, requestURL.getPort());
```
Comment (andreaTP) [inline components/http/okHttp/src/main/java/com/microsoft/kiota/http/OkHttpRequestAdapter.java:877]: since I was at it I changed this, I hope it's ok.

**Comment by andreaTP** [comment]
@baywet ready for review again 🙂

**Review by baywet — APPROVED** [review]
Thanks for making the changes!

**Pull request merged**